typescript/cli/src/config/hooks.ts

```diff
@@ -41,24 +41,55 @@ const MerkleTreeSchema = z.object({
   type: z.literal(HookType.MERKLE_TREE),
 });
 
-const HookSchema = z.union([ProtocolFeeSchema, MerkleTreeSchema]);
+const IGPSchema = z.object({
+  type: z.literal(HookType.INTERCHAIN_GAS_PAYMASTER),
+  owner: z.string(),
+  beneficiary: z.string(),
+  overhead: z.record(z.number()),
+  gasOracleType: z.record(z.literal(GasOracleContractType.StorageGasOracle)),
+  oracleKey: z.string(),
+});
+
+const RoutingConfigSchema: z.ZodSchema<any> = z.lazy(() =>
+  z.object({
+    type: z.literal(HookType.ROUTING),
+    owner: z.string(),
+    domains: z.record(HookConfigSchema),
+  }),
+);
+
+const AggregationConfigSchema: z.ZodSchema<any> = z.lazy(() =>
+  z.object({
+    type: z.literal(HookType.AGGREGATION),
+    hooks: z.array(HookConfigSchema),
+  }),
+);
 
-const ConfigSchema = z.object({
-  required: HookSchema,
-  default: HookSchema,
+const HookConfigSchema = z.union([
+  ProtocolFeeSchema,
+  MerkleTreeSchema,
+  IGPSchema,
+  RoutingConfigSchema,
+  AggregationConfigSchema,
+]);
+export type ZODHookConfig = z.infer<typeof HookConfigSchema>;
```

Comment:
Similar to the comment @yorhodes and I left in your other PR the other day about `isZODISMConfig`, I don't think this is the right naming convention here. I recommend just `HookConfig` for this type. `ZODHookConfig` makes me think it's a config **for** Zod

Reply:
it's only a config for ZOD though

Reply:
Using HookConfig makes it ambiguous, you don't know if you're using the static ts type or the one meant for ZOD so imo it's a nice separation of names